src/protocols/secure_channel/CASESession.cpp

```diff
@@ -224,12 +233,29 @@ class CASESession::WorkHelper
         bool cancel = false;
         // Execute callback in background thread; data must be OK with this
         helper->mStatus = helper->mWorkCallback(helper->mData, cancel);
+
+        // Check the status and call OnWorkFailedCallback
```

Comment:
As noted above, I don't think we should call the callback in the case of normal failure of the work. That failure is reported back to the caller (session object) through normal operation of the WorkHelper, when scheduling works, and the caller has to handle cleanup there anyways.

Reply:
reverted this change!